plugin/src/org/aavso/tools/vstar/external/plugin/FlexibleTextFileFormatObservationSink.java

```diff
@@ -0,0 +1,219 @@
+		
+		//ChartPanel panel = mediator.getPlotPane(mediator.getAnalysisType()).getChartPanel();
+		// Always take the titles of the RAW_DATA pane.
+		ChartPanel panel = mediator.getPlotPane(AnalysisType.RAW_DATA).getChartPanel();
+		JFreeChart chart = panel.getChart();
+		XYPlot plot = chart.getXYPlot();
+		
+		String titleX = plot.getDomainAxis().getLabel();
+		String titleY = plot.getRangeAxis().getLabel();
+		String title = chart.getTitle().getText();
+		String titlePrefix = LocaleProps.get("LIGHT_CURVE") + " " + LocaleProps.get("FOR") + " ";
+		if (title.length() > titlePrefix.length() && titlePrefix.equals(title.substring(0, titlePrefix.length())))
+			title = title.substring(titlePrefix.length(), title.length());
+		else
+			title = "^" + title;
+
+		if (title != null) writer.write("#NAME=" + title + "\n");
+		if (titleX != null) writer.write("#TITLEX=" + titleX + "\n");
+		if (titleY != null) writer.write("#TITLEY=" + titleY + "\n");
+		
+		writer.write("\n");
+		
+//		if (plot.getRangeAxis().isInverted())
+//			writer.write("#RANGEINVERTED=N\n");
+//		else
+//			writer.write("#RANGEINVERTED=Y\n");
+//
+//		writer.write("\n");
+
+		// copied from DownloadFormatObservationSinkPlugin with modifications
+		for (ValidObservation ob : obs) {
+			// Exclude excluded observations from the output like in DownloadFormatObservationSinkPlugin
+			if (!ob.isExcluded()) {
+				writer.write(observationToString(ob, delimiter));
+			}
+			mediator.getProgressNotifier().notifyListeners(ProgressInfo.INCREMENT_PROGRESS);
+		}
+	}
+	
+	private String observationToString(ValidObservation ob, String delimiter) {
+		StringBuffer buf = new StringBuffer();
+		
+		buf.append(ob.getDateInfo().getJulianDay());
+		buf.append(delimiter);
+		
+		buf.append(ob.getMagnitude().isFainterThan() ? "<" : "");
+		buf.append(ob.getMagnitude().getMagValue());
+		buf.append(delimiter);
+		
+		double uncertainty = ob.getMagnitude().getUncertainty();
+		if (uncertainty > 0.0) {
+			buf.append(uncertainty);
+		}
+		buf.append(delimiter);
+		
+		buf.append(quoteForCSVifNeeded(ob.getBand().getShortName(), delimiter));
+		buf.append(delimiter);
+		
+		if (!isNullOrEmpty(ob.getObsCode())) {
+			buf.append(quoteForCSVifNeeded(ob.getObsCode(), delimiter));
+		}
+		buf.append(delimiter);
+		
+		if (ob.getComments() != null) {
+			buf.append(quoteForCSVifNeeded(ob.getComments(), delimiter));
+		}
+		buf.append(delimiter);
+		
+		if (ob.getValidationType() != null) {
+			buf.append(ob.getValidationType().getValflag());
+		}
+		buf.append(delimiter);
+		
+		buf.append(quoteForCSVifNeeded(!isNullOrEmpty(ob.getName()) ? ob.getName() : "Unknown", delimiter));
+		//buf.append(delimiter);
+		
+		buf.append("\n");
+		return buf.toString();
+	}
+	
+	private String quoteForCSV(String field) {
+		return "\"" + field.replace("\"", "\"\"") + "\"";
+	}
+
+	private String quoteForCSVifNeeded(String field, String delimiter) {
+		return field.contains(delimiter) ? quoteForCSV(field) : field;
```

Comment:
An opportunity for refactoring here since `ValidObservation` has `quoteForCSVifNeeded` and `quoteForCSV` now I think?